components/contract-components/contract-release-form/index.tsx

```diff
@@ -184,561 +164,199 @@ export const ContractReleaseForm: React.FC<ContractReleaseFormProps> = ({
-                  {...register("isDeployableViaFactory")}
-                  isChecked={isDeployableViaFactory}
-                />
-                <Flex gap={1} alignItems="left" flexDir={"column"}>
-                  <Heading size="label.lg" flexShrink={0}>
-                    Deployable via factory (optional)
-                  </Heading>
-                  <Text>
-                    Use a factory contract to deploy clones of your
-                    implementation contracts using your custom logic.
-                  </Text>
-                </Flex>
-              </Flex>
-              {isDeployableViaFactory && (
-                <>
-                  <Heading size="label.lg" mt={8}>
-                    Addresses of your factory contracts
-                  </Heading>
-                  <Text>
-                    Enter the addresses of your deployed factory contracts.
-                    These need to conform to the{" "}
-                    <Link href="https://portal.thirdweb.com/contracts/IContractFactory">
-                      IContractFactory interface.
-                    </Link>
-                  </Text>
-                  <SimpleGrid columns={{ base: 1, md: 2 }} gap={4} mt={8}>
-                    {SUPPORTED_CHAIN_IDS.map((chainId) => (
-                      <FormControl key={`factory${chainId}`}>
-                        <FormLabel flex="1">
-                          {SupportedChainIdToNetworkMap[chainId]}
-                        </FormLabel>
-                        <Input
-                          {...register(
-                            `factoryDeploymentData.factoryAddresses.${chainId}`,
-                          )}
-                          placeholder="0x..."
-                          disabled={isDisabled}
-                        />
-                      </FormControl>
-                    ))}
-                  </SimpleGrid>
-                </>
+              {fieldsetToShow === "landing" && contractSelection === "proxy" ? (
+                <Button
+                  onClick={() => setFieldsetToShow("proxy")}
+                  colorScheme="primary"
+                  isDisabled={
+                    !form.watch("version") && !form.watch("displayName")
+                  }
+                >
+                  Next
+                </Button>
+              ) : fieldsetToShow === "landing" &&
+                contractSelection === "factory" ? (
+                <Button
+                  onClick={() => setFieldsetToShow("factory")}
+                  colorScheme="primary"
+                  isDisabled={
+                    !form.watch("version") && !form.watch("displayName")
+                  }
+                >
+                  Next
+                </Button>
+              ) : fieldsetToShow !== "contractParams" &&
+                deployParams?.length > 0 ? (
+                <Button
+                  isDisabled={
+                    !form.watch("version") && !form.watch("displayName")
```

Comment:
should be `||` not `&&` - if *either* one is empty - next should be disabled

Reply:
fixed in https://github.com/thirdweb-dev/dashboard/pull/782/commits/4085423a51450a30723a9dad73401a21d3fdd846